mavros/src/plugins/manual_controls.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @brief ManualControls plugin
+ * @file manual_controls.cpp
+ * @author Matias Nitsche <mnitsche@dc.uba.ar>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <mavros_msgs/ManualControls.h>
+
+namespace mavplugin {
+/**
+ * @brief Manual Controls plugin
+ */
+class ManualControlsPlugin : public MavRosPlugin {
+public:
+  ManualControlsPlugin() :
+    rc_nh("~manual_controls"),
+    uas(nullptr)
+	{ };
+
+	void initialize(UAS &uas_)
+	{
+		uas = &uas_;
+
+    controls_pub = rc_nh.advertise<mavros_msgs::ManualControls>("controls", 10);
```

Comment:
Ok. I have only coded publishing the data sent by the autopilot anyway.

On Fri, Aug 28, 2015 at 5:02 PM, Vladimir Ermakov <notifications@github.com>
wrote:

> In mavros/src/plugins/manual_controls.cpp
> <https://github.com/mavlink/mavros/pull/377#discussion_r38238454>:
>
> > +namespace mavplugin {
> > +/**
> > + * @brief Manual Controls plugin
> > + */
> > +class ManualControlsPlugin : public MavRosPlugin {
> > +public:
> > +  ManualControlsPlugin() :
> > +    rc_nh("~manual_controls"),
> > +    uas(nullptr)
> > +	{ };
> > +
> > +	void initialize(UAS &uas_)
> > +	{
> > +		uas = &uas_;
> > +
> > +    controls_pub = rc_nh.advertise<mavros_msgs::ManualControls>("controls", 10);
>
> As i know QGC may use that message for sending joystick data. So it might
> be useful to add possibility to send that message.
>
> In that case need better name for topic. Something like ~manual_control/to
> and ~manual_control/from.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mavlink/mavros/pull/377/files#r38238454>.
>